### PR TITLE
Build usage string based on command definitions

### PR DESCRIPTION
**Pull request by gurevichmark**
#4526

Generate Usage output based on the commands defined in `api_config_info` structure.
With this change Usage does not need to be manually added each time new commands are added to `api_config_info` structure.

man page, unfortunately, still has to me manually changed.

Before:
```
   OpenBMC specific:
       rspconfig <noderange> [ipsrc|ip|netmask|gateway|hostname|vlan]
       rspconfig <noderange> dump [-l|--list] [-g|--generate] [-c|--clear {<id>|all}] [-d|--download <id>]
```

After
```
   OpenBMC specific:
       rspconfig <noderange> [ipsrc|ip|netmask|gateway|hostname|vlan]
       rspconfig <noderange> dump [-l|--list] [-g|--generate] [-c|--clear {<id>|all}] [-d|--download <id>]
       rspconfig <noderange> powerrestorepolicy
       rspconfig <noderange> powerrestorepolicy={always_on|restore|always_off}
       rspconfig <noderange> powersupplyredundency
       rspconfig <noderange> powersupplyredundency={enabled|disabled}
       rspconfig <noderange> autoreboot
       rspconfig <noderange> autoreboot={0|1}
```

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 1]
```diff
@@ -697,7 +697,19 @@ sub preprocess_request {
         @exargs = @$extrargs;
     }
     my $usage_string = xCAT::Usage->parseCommand($command, @exargs);
+
     if ($usage_string) {
+        # Insert config api defined OpenBMC specific usage
+        my $iDataplex_tag = "iDataplex specific:";
```
[the remark this answers is not in the document]
Comment (whowutwut) [reply xCAT-server/lib/xcat/plugins/openbmc.pm]: Talked with @gurevichmark suggested that we change the Usage.pm to have a section for rspcofig_openbmc

my $rspconfig_begin = "blah blah"
my $rspconfig_openbmc = "openbmc usage..."
my $rspconfig_end = "blah blah" 
my $rspconfig = $rsconfig_begin . " " . $rspconfig_openbmc . " " . $rspconfig_end

and then just display $rspconfig_openbmc for the Usage.

**Comment by gurevichmark** [comment]
To collapse synopsis for each attribute from 2 lines to one, we can do something like this.
Both are a little more confusing than 2 separate lines, in my opinion

`rspconfig <nodelist> {powersupplyredundancy | powersupplyredundancy={Enabled | Disabled}}`

OR 

`rspconfig <nodelist> powersupplyredundancy[={Enabled | Disabled}]`

**Comment by whowutwut** [comment]
👍  @gurevichmark  Ok , let's leave that alone

File: xCAT-client/pods/man1/rspconfig.1.pod [hunk 2]
```diff
@@ -28,6 +28,18 @@ B<rspconfig> I<noderange> {B<ipsrc>|B<ip>|B<netmask>|B<gateway>|B<hostname>|B<vl
 
 B<rspconfig> I<noderange> B<dump> [B<-l>|B<--list>] [B<-g>|B<--generate>] [B<-c>|B<--clear> {I<id> | B<all>}] [B<-d>|B<--download> {I<id> | B<all>}]
 
+B<rspconfig> I<noderange> B<powerrestorepolicy>
+
+B<rspconfig> I<noderange> B<powerrestorepolicy={always_on|restore|always_off}>
+
+B<rspconfig> I<noderange> B<powersupplyredundancy>
+
+B<rspconfig> I<noderange> B<powersupplyredundancy={enabled|disabled}>
```
Comment (whowutwut) [inline xCAT-client/pods/man1/rspconfig.1.pod]: @gurevichmark Really nit picking, but the other field, autoreboot lists 0|1 so to be consistent, we should do `disabled|enabled`...  :-)

**Comment by whowutwut** [comment]
@gurevichmark After the last commit, can we run a quick UT with the code against IPMI and OpenBMC nodes, just to make sure that we didn't degrade IPMI managed machines when running rspconfig against those nodes.

**Comment by gurevichmark** [comment]
UT runs:

* OpenBMC node `rspconfig`  (autogenerated usage)

```
[root@briggs01 xcat]# rspconfig mid05tor12cn18 -h
Usage:
   Common:
       rspconfig [-h|--help|-v|--version|-V|--verbose]
   OpenBMC specific:
       rspconfig <noderange> [ipsrc|ip|netmask|gateway|hostname|vlan]
       rspconfig <noderange> dump [-l|--list] [-g|--generate] [-c|--clear {<id>|all}] [-d|--download {<id>|all}]
       rspconfig <noderange> powerrestorepolicy
       rspconfig <noderange> powerrestorepolicy={always_on|restore|always_off}
       rspconfig <noderange> powersupplyredundancy
       rspconfig <noderange> powersupplyredundancy={disabled|enabled}
       rspconfig <noderange> autoreboot
       rspconfig <noderange> autoreboot={0|1}

[root@briggs01 xcat]#
```

* OpenBMC node `rflash`  (no autogenerated usage)

```
[root@briggs01 xcat]# rflash mid05tor12cn18 -h
Usage:
    rflash [ -h|--help|-v|--version]
    PPC (with HMC) specific:
        rflash <noderange> -p <rpm_directory> [--activate {concurrent | disruptive}] [-V|--verbose]
        rflash <noderange> {--commit | --recover} [-V|--verbose]
    PPC (using Direct FSP Management) specific:
        rflash <noderange> -p <rpm_directory> [--activate {disruptive|deferred}] [-d <data_directory>]
        rflash <noderange> [--commit | --recover] [-V|--verbose]
        rflash <noderange> [--bpa_acdl]
    OpenPOWER BMC specific (using IPMI):
        rflash <noderange> [<hpm_file_path>|-d <data_directory>] [-c|--check] [--retry=<count>] [-V]
        rflash <noderange> --recover <bmc_file_path>
    OpenPOWER OpenBMC specific:
        rflash <noderange> {[-c|--check] | [-l|--list]}
        rflash <noderange> <tar_file_path> {[-c|--check] | [-a|--activate] | [-u|--upload]}
        rflash <noderange> <tar_file_diretory> [-d] [--no-host-reboot]
        rflash <noderange> <image_id> {[-a|--activate] | [--delete]}
[root@briggs01 xcat]#
```

* No node specified `rspconfig`  (no autogenerated usage)

```
[root@briggs01 xCAT]# rspconfig -h
Usage:
   Common:
       rspconfig [-h|--help|-v|--version|-V|--verbose]
   BMC/MPA Common:
       rspconfig <noderange> [snmpdest|alert|community] [-V|--verbose]
       rspconfig <noderange> [snmpdest=<dest ip address>|alert=<on|off|en|dis|enable|disable>|community=<string>]
   BMC specific:
       rspconfig <noderange> [ip|netmask|gateway|backupgateway|garp|vlan]
       rspconfig <noderange> [garp=<number of 1/2 second>]
       rspconfig <noderange> [userid=<userid> username=<username> password=<password>]
   OpenBMC specific:
       rspconfig <noderange> [ipsrc|ip|netmask|gateway|hostname|vlan]
       rspconfig <noderange> dump [-l|--list] [-g|--generate] [-c|--clear {<id>|all}] [-d|--download <id>]
   iDataplex specific:
       rspconfig <noderange> [thermprofile]
       rspconfig <noderange> [thermprofile=<two digit number from chassis>]
   MPA specific:
:
:
:
```

File: xCAT-client/pods/man1/rspconfig.1.pod (hunk 2 above)
Comment (gurevichmark) [reply xCAT-client/pods/man1/rspconfig.1.pod]: Fixed

**Pull request merged**